### ISSUE TITLE
(Feature Request) Error out for non-existent folders

### ISSUE THREAD

**Issue by Nasafato** [postmanlabs/newman#1061]
**Problem I'm facing**
If I run Newman with command line arguments (such as URLs) for the collections, environment, and folder I want to test, it won't tell me if it's actually found the resources at those URLs.

Instead, if the collection you specified does not exist, or you mispelled the name of the folder, Newman will run and exit with a non-error status code. It will simply display that no tests have run.

For example, running something like (I've purposely mispelled the collections endpoint):
```
newman run https://api.getpostman.com/collectiasdfasdfons/{{collection_uid}}?apikey={{postman-api-key-here}}\
--environment https://api.getpostman.com/environments/{{environment_uid}}?apikey={{postman-api-key-here}}
```
won't actually display any errors.

**Feature Request**
It would be a great help if `newman` would display a 404 error or exit with status code of 1 when it can't find these resources.

**Background**
Right now, I'm relying on these error codes to tell me whether or not a Newman test run has failed or not. My API server is built in Go, and some of my tests spin up the server locally and execute Newman against it. 

**Example Scenario**
Suppose that people make changes to the synced collection. For example, they change the name of a folder from "Basic API Test" to "Basic API Workflow", or change the name of a collection. Inevitably, I've hard-coded some names into my test scripts, but if such a change occurs, my Newman tests will still technically pass (since there are no tests to run). Which could be a big problem if it means they make it past Continuous Integration.

**Comment by Nasafato** [comment]
Didn't realize someone also had an issue with collections #1055.

**Comment by kunagpal** [comment]
@Nasafato The fix for this bug is distinct from #1055, and will be dealt with accordingly. Thanks for letting us know! 😄

**Comment by deepakpathania** [comment]
@Nasafato This has been fixed in the recent release. The `--bail` option now accepts modifiers that can be used to achieve this functionality, currently for an invalid folder name(will be extended later for collection, environment, globals and data)

So you can specify an optional modifier like `folder` with bail, which basically means that if an invalid folder name is found, newman will exit with an error status code which can be used in your tests to check whether the specified folder was found or not. 

for eg. `newman run collection.json --folder invalidName --bail folder` ends the collection run directly with an error saying `Unable to find a folder or request: invalidName` 

Hope this helps. Feel free to reach out in case you have additional querries.
